jenkins/jenkinspipeline-set-up-hnsesb-job.txt

```diff
@@ -0,0 +1,188 @@
+// This pipeline is created to build the application using build config and then deploy using deployment config
+pipeline {
+	
+	parameters {
+		choice(name: 'ACTION', choices: ['SETUP', 'DELETE'], description: 'Required action. If DELETE is selected, this job will delete all references of Application in selected namespace')
+		
+		choice(name: 'NAMESPACE', choices: ['Dev', 'Test'], description: 'Project namespace in openshift where you want to create this application')
+		
+        string(name: 'BRANCH_NAME', defaultValue: '3946-jenkins-job-for-new-hnsesb-insntance', description: 'Name of the branch')
+
+        text(name: 'Application', defaultValue: 'hnsesb-sample', description: 'Enter the name of application. This name will be used to define the URL. For example URL be: http://hnsesb-sample.apps.silver.devops.gov.bc.ca/ if value is hnsesb-sample. If application with same name already exist, this job will delete that application')
+
+		string(name: 'PHARMANET_CERT_ID', defaultValue: 'pharmanet-dev-secret', description: 'ID for Pharmanet cert in Jenkins')
+		
+		string(name: 'DATABASE_NAME', defaultValue: 'hnsesb-sample', description: 'Define database name for auditing database')
+		
+		string(name: 'DATABASE_USER', defaultValue: 'sample-user', description: 'Define database user for auditing database')
+		
+		password(name: 'DATABASE_PASSWORD', defaultValue: '', description: 'Set password for database')
+		
+ 		// Add the custom properties 
+		
+    }
+
+
+	environment {
+		CLUSTER = "openshift"
+    	OC_JENKINS_USER = "deployer-c5839f-tools"
+    	APP_NAME = lower()
+    	IMAGE = "hnsesb"
+    	TAG = "${APP_NAME}"
+		PROJECT = project()
+		PHARMANET_CREDENTIALS = credentials('pharmanet-dev')
```

Comment:
Would it make sense to have this configurable (but defaulted as well)? I.e. the user selects the credentials id to use in the Build screen? Same goes for the PHARMANET_CERT_PASSWORD. Might also be good to do for the DB USER/PASS.